superset/assets/javascripts/explore/stores/store.js

```diff
@@ -81,6 +81,9 @@ export function getControlsState(state, form_data) {
     control.value = formData[k] !== undefined ? formData[k] : control.default;
     controlsState[k] = control;
   });
+  if (viz.onInit) {
+    viz.onInit(controlsState);
```

Comment:
shouldn't we just `return viz.onInit(controlState)` Then in the function we can just return a new object:
```
onInit: (controlState) => {
    return {
      ...controlState
      time_grain_sqla: {
          value: null
      }
      granularity: {
          value: null
      }
   }
}
```

Reply:
gets a bit more verbose I think:
```javascript
onInit: (controlState) => {
    return {
      ...controlState
      time_grain_sqla: {
          ...controlState.time_grain_sqla,
          value: null,
      }
      granularity: {
          ...controlState.granularity,
          value: null,
      }
   }
}
```
it's clearly more functional / immutable, let me change it